src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/LocalAppContextSwitches.cs

```diff
@@ -18,21 +18,11 @@ private static bool InitializeDefaultActivityIdFormat()
                 string? switchValue = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_DIAGNOSTICS_DEFAULTACTIVITYIDFORMATISHIERARCHIAL");
                 if (switchValue != null)
                 {
-                    defaultActivityIdFormatIsHierarchial = IsTrueStringIgnoreCase(switchValue) || switchValue.Equals("1");
+                    defaultActivityIdFormatIsHierarchial = switchValue.Equals("true", StringComparison.OrdinalIgnoreCase) || switchValue.Equals("1");
```

Comment:
Afair, this will crash mono - https://github.com/dotnet/runtime/pull/77398#issuecomment-1292041726

Mono has to "fix" https://github.com/dotnet/runtime/issues/77513 to avoid that

Reply:
While you're here - can you please add a comment/TODO here?

Reply:
This isn't in corelib, this is in System.Diagnostics.DiagnosticSource.dll.  You're saying anyone who uses string.Equals in a static ctor in an arbitrary assembly can crash mono?  Or is there something special about DiagnosticSource as well?

Reply:
Ah I probably confused this with the corelib one - let's see what CI thinks 🙂